providers/windows/process_windows.go

```diff
@@ -61,51 +82,198 @@ func newProcess(pid int) (*process, error) {
 	return p, nil
 }
 
-func (p *process) init() error {
-	if p.pid != selfPID {
-		// TODO implement support for other processes.
-		return types.ErrNotImplemented
+// listProcesses returns a map of PID => PPID
+func listProcesses() (pids map[int]int, err error) {
+	snapHandle, err := syscall.CreateToolhelp32Snapshot(syscall.TH32CS_SNAPPROCESS, 0)
```

Comment:
I discarded EnumProcesses() because it didn't give me access to the parent pid (I didn't know about the undocumented field your mention below). Also, I read [some comments](https://stackoverflow.com/questions/4021307/enumprocesses-vs-createtoolhelp32snapshot) about EnumProcesses not listing all the processes under certain conditions. Now I checked and can confirm those claims are wrong.

However, there's not a big difference between EnumProcesses/CreateToolhelp32Snapshot. In my tests the later is about 4-5 ms slower, independent on the number of processes being listed.

Anyway, I switched to EnumProcesses now that I can get the parent in a more efficient way.